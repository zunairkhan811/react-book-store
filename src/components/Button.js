import PropTypes from 'prop-types';
import React from 'react';

function Button({
  type, className, onClick, value,
}) {
  return (
    <button
      type={type === 'submit' ? 'submit' : 'button'}
      className={className}
      onClick={onClick}
      value={value}
    >
      {value}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
export default Button;
