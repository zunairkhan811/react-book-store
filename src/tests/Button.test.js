import React from 'react';
import { render } from '@testing-library/react';
import Button from '../components/Button';

describe('Button Component', () => {
  it('should render Button component', () => {
    const props = {
      type: 'button',
      className: 'my-button',
      onClick: jest.fn(),
      value: 'Click me',
    };

    const { container } = render(<Button {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should render Button component with default onClick', () => {
    const props = {
      type: 'submit',
      className: 'my-button',
      value: 'Submit',
    };

    const { container } = render(<Button {...props} />);
    expect(container).toMatchSnapshot();
  });
});
