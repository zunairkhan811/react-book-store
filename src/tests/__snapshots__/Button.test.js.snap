// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render Button component 1`] = `
<div>
  <button
    class="my-button"
    type="button"
    value="Click me"
  >
    Click me
  </button>
</div>
`;

exports[`Button Component should render Button component with default onClick 1`] = `
<div>
  <button
    class="my-button"
    type="submit"
    value="Submit"
  >
    Submit
  </button>
</div>
`;
